pkg/agent/multicast/mcast_route_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build linux
+// +build linux
+
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicast
+
+import (
+	"net"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/tools/cache"
+
+	"antrea.io/antrea/pkg/agent/config"
+	multicasttest "antrea.io/antrea/pkg/agent/multicast/testing"
+)
+
+var (
+	addrIf1    = &net.IPNet{IP: nodeIf1IP, Mask: net.IPv4Mask(255, 255, 255, 0)}
+	addrIf2    = &net.IPNet{IP: externalInterfaceIP, Mask: net.IPv4Mask(255, 255, 255, 0)}
+	nodeConfig = &config.NodeConfig{GatewayConfig: &config.GatewayConfig{Name: "antrea-gw0"}, NodeIPv4Addr: addrIf1}
+)
+
+func TestParseIGMPMsg(t *testing.T) {
+	mRoute := newMockMulticastRouteClient(t)
+	err := mRoute.initialize(t)
+	assert.Nil(t, err)
+	for _, m := range []struct {
+		msg    []byte
+		errStr string
+	}{
+		{msg: []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 11, 11, 12, 24, 12, 34, 21, 23, 45},
+			errStr: "failed to parse IGMPMSG: message length should greater than 19"},
+		{msg: []byte{1, 2, 3, 4, 5, 6, 7, 8, 0, 0, 11, 11, 12, 24, 12, 34, 21, 23, 45, 22},
+			errStr: "not a IGMPMSG_NOCACHE message: [1 2 3 4 5 6 7 8 0 0 11 11 12 24 12 34 21 23 45 22]"},
+	} {
+		_, err := mRoute.parseIGMPMsg(m.msg)
+		assert.NotNil(t, err)
+		assert.Equal(t, m.errStr, err.Error())
+	}
+
+	for _, m := range []struct {
+		msg           []byte
+		parsedIGMPMsg parsedIGMPMsg
+	}{
+		{msg: []byte{69, 0, 0, 28, 242, 241, 64, 0, 1, 0, 0, 0, 10, 244, 0, 2, 224, 3, 3, 7, 1, 0, 0, 0, 0, 0, 0, 0},
```

Comment:
Updated.  Minor difference from your comment: `assert.Error(t, tc.expectedErr, err)` won't work because `assert.Error` does not allow err to be nil. Changed it to `assert.Equal`.